crates/typst/src/foundations/args.rs

```diff
@@ -76,8 +76,20 @@ impl Args {
         self.items.iter().filter(|slot| slot.name.is_none()).count()
     }
 
-    /// Push a positional argument.
-    pub fn push(&mut self, span: Span, value: Value) {
+    /// Push a positional argument to the front.
+    pub fn push_front(&mut self, span: Span, value: Value) {
+        self.items.insert(
+            0,
+            Arg {
+                span: self.span,
+                name: None,
+                value: Spanned::new(value, span),
+            },
+        )
+    }
+
+    /// Push a positional argument to the back.
+    pub fn push_back(&mut self, span: Span, value: Value) {
```

Comment:
These names are slightly confusing due to the similarity with `VecDeque`. I'd suggest the following instead (while replacing earlier usage with `args.insert(0, span, value)`):

```suggestion
    /// Insert a positional argument at a specific index.
    pub fn insert(&mut self, index: usize, span: Span, value: Value) {
        self.items.insert(
            index,
            Arg {
                span: self.span,
                name: None,
                value: Spanned::new(value, span),
            },
        )
    }

    /// Push a positional argument.
    pub fn push(&mut self, span: Span, value: Value) {
```